File: 1_algorithmic_toolbox/week1_programming_challenges/2maxpp.py
# python3

if __name__ == '__main__':
    n = int(input())
    a = [int(i) for i in input().split()]

    max1, max2 = max(a[0], a[1]), min(a[0], a[1])

    for i in range(2, n):
        if a[i] >= max1:
            max2 = max1
            max1 = a[i]
        elif a[i] >= max2:
            max2 = a[i]

    print(max1 * max2)

File: 1_algorithmic_toolbox/week1_programming_challenges/2maxpp.cpp
#include <iostream>
#include <vector>

using namespace std;

long long MaxPairwiseProduct(const vector<int>& v) {
	int max1 = -1, max2 = -1;
	for (const auto& i : v) {
		if (i > max1) {
			max2 = max1;
			max1 = i;
		}
		else if (i > max2) {
			max2 = i;
		}
	}
	return (long long)max1 * (long long)max2;
}

int main() {
	int n;
	cin >> n;
	vector<int> v(n);
	for (auto& i : v) {
		cin >> i;
	}
	cout << MaxPairwiseProduct(v) << endl;
	return 0;
}
